lib/bolt/pal.rb

```diff
@@ -351,6 +368,17 @@ def list_modules
       end
     end
 
+    def generate_types(config)
+      require 'puppet/face/generate'
+      in_bolt_compiler do
+        generator = Puppet::Generate::Type
+        inputs = generator.find_inputs(:pcore)
+        outputdir = File.join(config.boltdir.path, '.resource_types')
```

Comment:
Should this just be `config.boltdir.resource_types`?

Reply:
Actually, now that pal has an attribute `@resource_types` I dont think this method even needs an argument.